[Java] 3.61.0 Release Notes (#627)
SAP/cloud-sdk: docs/java/release-notes.mdx

### Fixed issues
-->

## 3.61.0 - January 13, 2022

[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.61.0) | [Javadoc](https://help.sap.com/doc/d36e6a93f11b48bd920453bed2149bd3/1.0/en-US/index.html)

### New Functionality

> - The SAP Cloud SDK now supports a new `ProxyType` named `PrivateLink`.
>   This type will be present on destinations created for the [Private Link Service](https://help.sap.com/viewer/product/PRIVATE_LINK/CLOUD/en-US).
> - The SAP Cloud SDK now provides a new accessor for _HTTP headers_ of the _incoming request_.
>   The new `RequestHeaderAccessor` (located in the `cloudplatform-core` module) can be used to conveniently access the HTTP headers of the currently processed incoming request.
>   Furthermore, it can be used to also propagate specific headers to nested `ThreadContext`s, much like the other _accessors_ of the Cloud SDK.
>   By default, all present headers are automatically "inherited" when creating a new `ThreadContext` so that even asynchronous operations are able to access the initial headers.

Reviewer: ```suggestion
- The SAP Cloud SDK now supports a new `ProxyType` named `PrivateLink`.
  This proxy type is used when your destination represents a tunnel created via [Private Link Service](https://help.sap.com/viewer/product/PRIVATE_LINK/CLOUD/en-US).
- The SAP Cloud SDK now provides a new accessor for _HTTP headers_ of the _incoming request_.
  The new `RequestHeaderAccessor` located in the `cloudplatform-core` module can be used to conveniently access HTTP headers of the currently processed incoming request.
  Furthermore, it can be used to also propagate specific headers to nested `ThreadContext`s, much like the other _accessors_ of the Cloud SDK.
  By default, all present headers are automatically "inherited" when creating a new `ThreadContext` so that even asynchronous operations are able to access the initial headers.
```

        - Update [Json](https://search.maven.org/search?q=g:org.json%2BAND%2Ba:json) (`org.json:json`) from `20210307` to `20211205`
        - Update [Netty](https://search.maven.org/search?q=g:io.netty%2BAND%2Ba:netty-bom) (`io.netty:netty-bom`) from `4.1.70.Final` to `4.1.72.Final`
        - Update [Plexus Component Annotations](https://search.maven.org/search?q=g:org.codehaus.plexus%2BAND%2Ba:plexus-component-annotations) (`org.codehaus.plexus:plexus-component-annotations`) from `2.1.0` to `2.1.1`
        - Update [Reactor Core](https://search.maven.org/search?q=g:io.projectreactor%2BAND%2Ba:reactor-core) (`io.projectreactor:reactor-core`) from `3.4.12` to `3.4.13`
        - Update [System Lambda](https://search.maven.org/search?q=g:com.github.stefanbirkner%2BAND%2Ba:system-lambda) (`com.github.stefanbirkner:system-lambda`) from `1.2.0` to `1.2.1`
        - Update [Wiremock](https://search.maven.org/search?q=g:com.github.tomakehurst%2BAND%2Ba:wiremock-jre8-standalone) (`com.github.tomakehurst:wiremock-jre8-standalone`) from `2.31.0` to `2.32.0`

### Fixed issues

> - Fix a regression in version `3.51.0` that caused SOAP transaction commits to not be applied to the same session as the requests being committed.
> - Fix an issue that could sometimes lead to a `ConcurrentModificationException` when using the Destination or Connectivity service functionality.
> - Fix an issue that could sometimes lead to `CacheException Cache already exists` when using the Destination / Connectivity service or resilience functionality.
> - Fix an issue where service bindings were not loaded from the file system in case the working directory of the Java process was not the root directory.
> - Fix an issue where custom service-path information are not propagated from OData V2 service classes to batch classes.
>   Previously this let to a runtime exception: `ODataRequestException: Batch request contains requests to different service paths [...]`.
>   To enable the fix, please use the latest [OData V2 VDM](https://sap.github.io/cloud-sdk/docs/java/features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java) and/or [re-generate your own OData V2 classes](https://sap.github.io/cloud-sdk/docs/java/features/odata/generate-typed-odata-v2-and-v4-client-for-java).
> - Fix [CVE-2021-33813](https://nvd.nist.gov/vuln/detail/CVE-2021-33813) by upgrading `jdom2` to version `2.0.6.1`

Reviewer: ```suggestion
- Fix a regression in version `3.51.0` that caused SOAP transaction commits to not be applied to the same session as the requests being committed.
- Fix an issue that could sometimes lead to a `ConcurrentModificationException` when using the Destination or Connectivity service functionality.
- Fix an issue that could sometimes lead to `CacheException Cache already exists` when using the Destination / Connectivity service or resilience functionality.
- Fix an issue where service bindings were not loaded from the file system in case the working directory of the Java process was not the root directory.
- Fix an issue where custom service-path information is not propagated from OData V2 service classes to batch classes.
  Previously this led to a runtime exception: `ODataRequestException: Batch request contains requests to different service paths [...]`.
  To enable the fix, please use the latest [OData V2 VDM](https://sap.github.io/cloud-sdk/docs/java/features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java) and/or [re-generate your own OData V2 classes](https://sap.github.io/cloud-sdk/docs/java/features/odata/generate-typed-odata-v2-and-v4-client-for-java).
- Fix [CVE-2021-33813](https://nvd.nist.gov/vuln/detail/CVE-2021-33813) by upgrading `jdom2` to version `2.0.6.1`
```
